extensions/resource-deployment/src/ui/modelViewUtils.ts

```diff
@@ -256,24 +295,41 @@ function processFields(fieldInfoArray: FieldInfo[], components: azdata.Component
 	}
 }
 
-export function createFlexContainer(view: azdata.ModelView, items: azdata.Component[], rowLayout: boolean = true): azdata.FlexContainer {
+export function createFlexContainer(view: azdata.ModelView, items: azdata.Component[], rowLayout: boolean = true, width?: string | number, height?: string | number, alignItems?: azdata.AlignItemsType, cssStyles?: CSSStyles): azdata.FlexContainer {
 	const flexFlow = rowLayout ? 'row' : 'column';
-	const alignItems = rowLayout ? 'center' : undefined;
-	const itemsStyle = rowLayout ? { CSSStyles: { 'margin-right': '5px' } } : {};
-	return view.modelBuilder.flexContainer().withItems(items, itemsStyle).withLayout({ flexFlow: flexFlow, alignItems: alignItems }).component();
+	alignItems = alignItems || (rowLayout ? 'center' : undefined);
+	const itemsStyle = rowLayout ? { CSSStyles: { 'margin-right': '5px', } } : {};
+	const flexLayout: azdata.FlexLayout = { flexFlow: flexFlow };
+	if (height) {
+		flexLayout.height = height;
+	}
+	if (width) {
+		flexLayout.width = width;
+	}
+	if (alignItems) {
+		flexLayout.alignItems = alignItems;
+	}
+	let flexComponent = view.modelBuilder.flexContainer().withItems(items, itemsStyle).withLayout(flexLayout).component();
+	if ( cssStyles ) {
+		flexComponent.CSSStyles = Object.assign(flexComponent.CSSStyles || {}, cssStyles);
+	}
+	return flexComponent;
 }
 
 export function createGroupContainer(view: azdata.ModelView, items: azdata.Component[], layout: azdata.GroupLayout): azdata.GroupContainer {
-	return view.modelBuilder.groupContainer().withItems(items).withLayout(layout).component();
+	return view.modelBuilder.groupContainer().withItems(items).	withLayout(layout).component();
 }
 
-function addLabelInputPairToContainer(view: azdata.ModelView, components: azdata.Component[], label: azdata.Component, input: azdata.Component, labelPosition?: LabelPosition, additionalComponents?: azdata.Component[]) {
-	const inputs = [label, input];
+function addLabelInputPairToContainer(view: azdata.ModelView, components: azdata.Component[], label: azdata.Component, input: azdata.Component | undefined, fieldInfo: FieldInfo, additionalComponents?: azdata.Component[]) {
+	const inputs: azdata.Component[] = [label];
+	if (input !== undefined) {
+		inputs.push(input);
+	}
 	if (additionalComponents && additionalComponents.length > 0) {
 		inputs.push(...additionalComponents);
 	}
-	if (labelPosition && labelPosition === LabelPosition.Left) {
-		const row = createFlexContainer(view, inputs);
+	if (inputs.length > 0 && fieldInfo.labelPosition && fieldInfo.labelPosition === LabelPosition.Left) {
```

Comment:
Inputs.length is always going to be > 0 since we start off with the label component in there

And you can just shorten this to fieldInfo.labelPosition === LabelPosition.Left, if it's undefined then it won't be equal anyways

Reply:
done